pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -214,10 +225,101 @@ func (h *AppHandler) ProduceArtifacts(ctx context.Context, comps []*types.Compon
 	return h.createResourcesConfigMap(ctx, h.currentAppRev, comps, policies)
 }
 
+// collectPartialHealthStatus will collect the health status for partial manifests
+func (h *AppHandler) collectPartialHealthStatus(ctx context.Context, wl *appfile.Workload, appRev *v1beta1.ApplicationRevision, options DispatchOptions) (bool, error) {
```

Comment:
So you copied the collectHealthStatus function here?

Reply:
suggest make some common logic into separated functions for re-use

Reply:
I will refactor this part of the logic.